updated QML intro based on Alex's edits (#996)
MicrosoftDocs/quantum-docs-pr: articles/user-guide/libraries/machine-learning/intro.md

## Framework and goals

> Quantum encoding and processing of information is a powerful alternative to classical machine learning

Reviewer: It looks like some punctuation is missing here?

> Quantum classifiers. In particular, it allows to encode data in quantum registers that are concise relative to the number of features, systematically employing quantum entanglement as computational resource and employing quantum measurement for class inference.

Reviewer: It looks like this is a sentence fragment now?

> Quantum classifiers. In particular, it allows to encode data in quantum registers that are concise relative to the number of features, systematically employing quantum entanglement as computational resource and employing quantum measurement for class inference.

Reviewer: ```suggestion
Quantum classifiers. In particular, it allows us to encode data in quantum registers that are concise relative to the number of features, systematically employing quantum entanglement as computational resource and employing quantum measurement for class inference.
```

- data encoding
- preparation of a classifier state
- measurement
Due to the probabilistic nature of the measurement, these three steps must be repeated multiple times. 
Both the encoding and the computing of the classifier state are done by means of *quantum circuits*. While the encoding circuit is usually data-driven and parameter-free, the classifier circuit contains a sufficient set of learnable parameters. 

In the proposed solution the classifier circuit is composed of single-qubit rotations and two-qubit controlled rotations. The learnable parameters here are the rotation angles. The rotation and controlled rotation gates are known to be *universal* for quantum computation, which means that any unitary weight matrix can be decomposed into a long enough circuit consisting of such gates.

> In the proposed version only one circuit followed by a single frequency estimation is supported, which makes the solution a quantum analog of support vector machine with low-degree polynomial kernel.

Reviewer: ```suggestion
In the proposed version, only one circuit followed by a single frequency estimation is supported.
Thus, the solution is a quantum analog of a support vector machine with a low-degree polynomial kernel.
```

This could also be a good place to refer further resources on the kernel trick, since we use it implicitly quite a bit in this argument.

![Multilayer perceptron vs. Circuit Centric Classifier](~/media/DLvsQCC.png)

> A simple quantum classifier design can be compared to a traditional support vector machine (SVM) solution. The inference for a data sample $x$ in case of SVM is done using an optimal kernel form $\sum \alpha_j  k(x_j,x)$ where $k$ is a certain kernel function. Quantum classifier uses the predictor $p(y│x,U(\theta))=〈U(\theta)x|M|U(\theta)x〉$ which is similar in spirit but technically quite different. Thus when straightforward amplitude encoding is used,  $p(y│x,U(\theta))$ is a quadratic form in the amplitudes of $x$, but the coefficients of this form are no longer learned independently; they are instead aggregated from the matrix elements of the circuit $U(\theta)$, which typically has significantly fewer learnable parameters $\theta$ than the dimension of the vector $x$. The polynomial degree of $p(y│x,U(\theta))$ in the original features can be increased to $2^l$ by using quantum product encoding on $l$ copies of $x$.

Reviewer: As a side note, breaking these down into smaller lines can help make more readable / commentable diffs.

> A simple quantum classifier design can be compared to a traditional support vector machine (SVM) solution. The inference for a data sample $x$ in case of SVM is done using an optimal kernel form $\sum \alpha_j  k(x_j,x)$ where $k$ is a certain kernel function. Quantum classifier uses the predictor $p(y│x,U(\theta))=〈U(\theta)x|M|U(\theta)x〉$ which is similar in spirit but technically quite different. Thus when straightforward amplitude encoding is used,  $p(y│x,U(\theta))$ is a quadratic form in the amplitudes of $x$, but the coefficients of this form are no longer learned independently; they are instead aggregated from the matrix elements of the circuit $U(\theta)$, which typically has significantly fewer learnable parameters $\theta$ than the dimension of the vector $x$. The polynomial degree of $p(y│x,U(\theta))$ in the original features can be increased to $2^l$ by using quantum product encoding on $l$ copies of $x$.

Reviewer: ```suggestion
A simple quantum classifier design can be compared to a traditional support vector machine (SVM) solution. The inference for a data sample $x$ in case of SVM is done using an optimal kernel form $\sum \alpha_j  k(x_j,x)$ where $k$ is a certain kernel function.
By contrast, a quantum classifier uses the predictor $p(y│x,U(\theta))=〈U(\theta)x|M|U(\theta)x〉$, which is similar in spirit but technically quite different. Thus, when a straightforward amplitude encoding is used,  $p(y│x,U(\theta))$ is a quadratic form in the amplitudes of $x$, but the coefficients of this form are no longer learned independently; they are instead aggregated from the matrix elements of the circuit $U(\theta)$, which typically has significantly fewer learnable parameters $\theta$ than the dimension of the vector $x$. The polynomial degree of $p(y│x,U(\theta))$ in the original features can be increased to $2^l$ by using a quantum product encoding on $l$ copies of $x$.
```

- data encoding
- preparation of a classifier state
- measurement
Due to the probabilistic nature of the measurement, these three steps must be repeated multiple times. 
Both the encoding and the computing of the classifier state are done by means of *quantum circuits*. While the encoding circuit is usually data-driven and parameter-free, the classifier circuit contains a sufficient set of learnable parameters. 

In the proposed solution the classifier circuit is composed of single-qubit rotations and two-qubit controlled rotations. The learnable parameters here are the rotation angles. The rotation and controlled rotation gates are known to be *universal* for quantum computation, which means that any unitary weight matrix can be decomposed into a long enough circuit consisting of such gates.

In the proposed version, only one circuit followed by a single frequency estimation is supported.
Thus, the solution is a quantum analog of a support vector machine with a low-degree polynomial kernel.

![Multilayer perceptron vs. Circuit Centric Classifier](~/media/DLvsQCC.png)

> A simple quantum classifier design can be compared to a traditional support vector machine (SVM) solution. The inference for a data sample $x$ in case of SVM is done using an optimal kernel form $\sum \alpha_j  k(x_j,x)$ where $k$ is a certain kernel function.
> By contrast, a quantum classifier uses the predictor $p(y│x,U(\theta))=〈U(\theta)x|M|U(\theta)x〉$, which is similar in spirit but technically quite different. Thus, when a straightforward amplitude encoding is used,  $p(y│x,U(\theta))$ is a quadratic form in the amplitudes of $x$, but the coefficients of this form are no longer learned independently; they are instead aggregated from the matrix elements of the circuit $U(\theta)$, which typically has significantly fewer learnable parameters $\theta$ than the dimension of the vector $x$. The polynomial degree of $p(y│x,U(\theta))$ in the original features can be increased to $2^l$ by using a quantum product encoding on $l$ copies of $x$.

Author: ```suggestion
A simple quantum classifier design can be compared to a traditional support vector machine (SVM) solution. The inference for a data sample $x$ in case of SVM is done using an optimal kernel form $\sum \alpha_j  k(x_j,x)$ where $k$ is a certain kernel function.

By contrast, a quantum classifier uses the predictor $p(y│x,U(\theta))=〈U(\theta)x|M|U(\theta)x〉$, which is similar in spirit but technically quite different. Thus, when a straightforward amplitude encoding is used,  $p(y│x,U(\theta))$ is a quadratic form in the amplitudes of $x$, but the coefficients of this form are no longer learned independently; they are instead aggregated from the matrix elements of the circuit $U(\theta)$, which typically has significantly fewer learnable parameters $\theta$ than the dimension of the vector $x$. The polynomial degree of $p(y│x,U(\theta))$ in the original features can be increased to $2^l$ by using a quantum product encoding on $l$ copies of $x$.
```
